feat: implement CE/EE architecture

flemzord opened the pull request.

flemzord committed:
feat: implement CE/EE architecture

flemzord committed:
feat: Add Audit configuration

gfyrag commented on .goreleaser.yml:
```diff
@@ -7,8 +7,9 @@ includes:
       url: https://raw.githubusercontent.com/formancehq/earthly/refs/heads/main/.goreleaser.default.yaml
 
 builds:
+  # Community Edition build
```
I guess changes in this file can break a lot of user automation.

gfyrag commented on go.mod:
```diff
@@ -51,7 +51,7 @@ require (
 require (
 	github.com/ClickHouse/clickhouse-go/v2 v2.37.2
-	github.com/formancehq/go-libs/v3 v3.4.0
+	github.com/formancehq/go-libs/v3 v3.5.1-0.20251107100055-6f51cf04fd2e
```
Use a tag version before merge.

flemzord committed:
fix: All working

flemzord commented on cmd/serve.go:
```diff
@@ -186,6 +188,9 @@ func NewServeCommand() *cobra.Command {
 	cmd.Flags().Bool(SemconvMetricsNames, false, "Use semconv metrics names (recommended)")
 
+	// Add EE flags (if EE build)
+	ee.AddFlags(cmd)
```
<img width="1622" height="197" alt="CleanShot 2025-11-07 at 13 47 56" src="https://github.com/user-attachments/assets/cc80be73-9d15-458d-aace-9da36a8b6bf7" />
Yes, that's exactly the case

flemzord committed:
fix: resolve import cycle and add audit middleware properly

- Fix import cycle between ee and cmd packages by passing serviceName as parameter
- Move audit middleware initialization before route mounting to fix Chi panic
- Add 'ee' build tag to linting for tools and pulumi packages
- Remove duplicate serviceName constant from ee/audit.go

This ensures the audit middleware is added before routes are defined,
preventing the "all middlewares must be defined before routes" panic.

flemzord committed:
fix: don't change the default id in goreleaser

flemzord committed:
fix: update archive and file name templates to remove 'ce' suffix

flemzord committed:
fix: remove unused Community Edition stubs in ee package

flemzord committed:
fix: handle errors when loading audit settings in AuditModule

flemzord committed:
fix: handle error when reading audit enabled flag in AuditModule
